From 6efe7d7dc3e105a1ea19042f6b212b3366e4fd13 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@strace.io>
Date: Wed, 12 Jan 2022 08:00:00 +0000
Subject: [PATCH 2/5] Remove unused function syscall_name

The last user of this function was removed by commit v5.15~209.

* src/defs.h (syscall_name): Remove.
* src/syscall_name.c: Likewise.
---
 src/defs.h         | 8 --------
 src/syscall_name.c | 6 ------
 2 files changed, 14 deletions(-)

diff --git a/src/defs.h b/src/defs.h
index b95f6bdd3..8f5fab487 100644
--- a/src/defs.h
+++ b/src/defs.h
@@ -577,14 +577,6 @@ extern void clear_regs(struct tcb *tcp);
 extern int get_scno(struct tcb *);
 extern kernel_ulong_t get_rt_sigframe_addr(struct tcb *);
 
-/**
- * Convert a (shuffled) syscall number to the corresponding syscall name.
- *
- * @param scno Syscall number.
- * @return     String literal corresponding to the syscall number in case latter
- *             is valid; NULL otherwise.
- */
-extern const char *syscall_name(kernel_ulong_t scno);
 /**
  * Convert a pair of (raw) syscall number and arch (defined by an AUDIT_ARCH_*
  * constant) to the corresponding syscall name.  So far works only for arches
diff --git a/src/syscall_name.c b/src/syscall_name.c
index 72ba85bc3..bb0bd2ce0 100644
--- a/src/syscall_name.c
+++ b/src/syscall_name.c
@@ -34,12 +34,6 @@ const struct audit_arch_t audit_arch_vec[SUPPORTED_PERSONALITIES] = {
 };
 
 
-const char *
-syscall_name(kernel_ulong_t scno)
-{
-	return scno_is_valid(scno) ? sysent[scno].sys_name : NULL;
-}
-
 const char *
 syscall_name_arch(kernel_ulong_t nr, unsigned int arch, const char **prefix)
 {
-- 
2.34.1

